src/gurt/misc.c

```diff
@@ -35,31 +53,61 @@ d_rand()
 void
 d_free(void *ptr)
 {
+	size_t size = malloc_usable_size(ptr);
+
+	if (d_free_track_cb != NULL && track_arg != NULL)
+		d_free_track_cb(track_arg, size);
+
 	free(ptr);
 }
 
 void *
 d_calloc(size_t count, size_t eltsize)
 {
-	return calloc(count, eltsize);
+	void *ptr;
+
+	ptr = calloc(count, eltsize);
+	if (ptr != NULL && d_alloc_track_cb != NULL && track_arg != NULL)
+		d_alloc_track_cb(track_arg, eltsize);
```

Comment:
The real allocated size may be greater than the required "eltsize" because of alignment. So the "eltsize" maybe not equal to the size you will get via malloc_usable_size() when free the ptr. That will confuse the metrics. I think that we need to always use malloc_usable_size() in spite of when alloc or free for metrics purpose. Similar for other cases.

Reply:
Ah, right it should be be eltsize * count. 

Actually "The value returned by malloc_usable_size() may be greater than the requested size of the allocation because of alignment and minimum size constraints." 
mostly it does not need to be precisely, but malloc_usable_size() should be good enough anyway.  so we can know the about memory usage on the current system.

And also there is overhead for malloc_usable_size() I assume, so I have to avoid to use that too often as much as possible.  Though I do not have other choice for d_free().